src/neptune_scale/core/components/sync_process.py

```diff
@@ -367,7 +367,11 @@ def commit(self) -> None:
 
     def work(self) -> None:
         try:
-            while (operation := self.get_next()) is not None:
+            while not self._is_interrupted():
```

Comment:
With Python 3.8 assumption, I think it's possible to to use `while not self._is_interrupted() and (operation := self.get_next()):` instead

Reply:
I don't want to break the loop though, just keep blocking on the queue without any sleeps between calls to `work()`.